pkg/webui/console/containers/application-payload-formatters/downlink.js

```diff
@@ -52,115 +50,78 @@ const m = defineMessages({
   downlinkResetWarning:
     'You do not have sufficient rights to view the current downlink payload formatter. Only overwriting is allowed.',
 })
-@connect(
-  state => {
-    const formatters = selectApplicationLinkFormatters(state) || {}
-    const mayViewLink = checkFromState(mayViewApplicationLink, state)
-    const linkError = selectApplicationLinkError(state)
-
-    return {
-      appId: selectSelectedApplicationId(state),
-      formatters,
-      mayViewLink,
-      linkError,
-    }
-  },
-  {
-    updateLinkSuccess: updateApplicationLinkSuccess,
-    submitPayloadFormatters: (id, pf) => attachPromise(updateApplicationLink(id, pf)),
-  },
-)
-@withBreadcrumb('apps.single.payload-formatters.downlink', props => {
-  const { appId } = props
 
-  return (
+const ApplicationPayloadFormatters = () => {
+  const appId = useSelector(selectSelectedApplicationId)
```

Comment:
```suggestion
  const { appId } = useParams()
```
Let's already switch to parameters here. Once we're done with all refactors I also want to get rid of selected entity IDs in the store, which we don't need to store anymore if we can access the route parameter from anywhere.